packages/enzyme/src/ReactWrapper.js

```diff
@@ -104,20 +107,33 @@ class ReactWrapper {
         throw new TypeError('ReactWrapper can only wrap valid elements');
       }
 
-      privateSet(this, UNRENDERED, nodes);
       const renderer = adapter.createRenderer({ mode: 'mount', ...options });
       privateSet(this, RENDERER, renderer);
       renderer.render(nodes, options.context);
       privateSet(this, ROOT, this);
       privateSetNodes(this, this[RENDERER].getNode());
     } else {
-      privateSet(this, UNRENDERED, null);
       privateSet(this, RENDERER, root[RENDERER]);
       privateSet(this, ROOT, root);
       privateSetNodes(this, nodes);
       privateSet(this, ROOT_NODES, root[NODES]);
     }
+    privateSet(this, UNRENDERED, nodes);
     privateSet(this, OPTIONS, root ? root[OPTIONS] : options);
+    privateSet(this, LINKED_ROOTS, []);
+    privateSet(this, UPDATED_BY, null);
+
+    if (typeof options.wrappingComponent === 'function') {
```

Comment:
should maybe this be a function also defined on adapters, so that most of them can use `react-is`, and the react 16 ones will be able to support non-function component types, like forwardRef?

Reply:
Good call!